src/app/auth/signup.tsx

```diff
@@ -44,18 +87,78 @@ function SignUpScreen() {
         <Input
           label="Password"
           leftIcon={{ type: 'font-awesome', name: 'lock' }}
-          onChangeText={text => setPassword(text)}
+          onChangeText={text => checkPassword(text)}
           value={password}
           secureTextEntry
           placeholder="Password"
           autoCapitalize="none"
         />
+      </View>
+      <View>
+        {password !== '' && (
+          <View style={globalStyles.passwordComplexity}>
+            <Icon type={hasUppercase ? 'green_check' : 'red_x'} />
+            <Text
+              style={
+                hasUppercase
+                  ? { color: colors.textGreen }
+                  : { color: colors.textRed }
+              }
+            >
+              At least 1 uppercase letter.
+            </Text>
+          </View>
+        )}
+        {password !== '' && (
+          <View style={globalStyles.passwordComplexity}>
+            <Icon type={hasLowercase ? 'green_check' : 'red_x'} />
+            <Text
+              style={
+                hasLowercase
+                  ? { color: colors.textGreen }
```

Comment:
Nit: looking at the designs, we want this fontSize to be 12 and there to be a gap of 8 between the icon and the text. To do this, we can replace these lines with 
` style={[
              styles.errorText,
              hasLength
                ? { color: colors.textGreen }
                : { color: colors.textRed },
            ]}
`
where the styles for errorText include fontSize: 12 and marginLeft: 8